app/controllers/users/sessions_controller.rb

```diff
@@ -1,11 +1,19 @@
 # frozen_string_literal: true
 
 class Users::SessionsController < Devise::SessionsController
+  include ApplicationHelper
+
   class EmailNotFoundError < StandardError; end
   class LoginIncompleteError < StandardError; end
 
   def create
-    super
+    if Rails.env.development? || Rails.env.deployed_development?
+      user = User.find_by_email(params.dig(:user, :email))
+      sign_in(user, scope: :user)
+      redirect_to profile_dashboard_url(user)
+    else
+      super
```

Comment:
Can we do something more ugly here? Like, check if the email of the user is in the list of permitted test values, check if we are in dev / deployed dev, and only then log them in without email?

I think that's the bit that we discussed yesterday - so that our test accounts get in without fuss, but login mechanism still works for other users (in particular, I would keep `ecf@mailinator.com` as not automatically logged in).

Reply:
Where do you think would be a good place to maintain that list of emails?

Reply:
Wherever convenient and not disgusting.

Constant in the controller is fine by me.